backend/molgenis-emx2-analytics/src/main/java/org/molgenis/emx2/analytics/repository/TriggerRepositoryImpl.java

```diff
@@ -0,0 +1,192 @@
+package org.molgenis.emx2.analytics.repository;
+
+import static org.molgenis.emx2.Column.column;
+import static org.molgenis.emx2.Constants.SYSTEM_SCHEMA;
+import static org.molgenis.emx2.FilterBean.f;
+import static org.molgenis.emx2.Operator.EQUALS;
+import static org.molgenis.emx2.Row.row;
+import static org.molgenis.emx2.TableMetadata.table;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.molgenis.emx2.*;
+import org.molgenis.emx2.analytics.model.Trigger;
+
+public class TriggerRepositoryImpl implements TriggerRepository {
+  static final String TRIGGER_TABLE_NAME = "AnalyticsTrigger";
+
+  private static final String NAME = "name";
+  private static final String CSS_SELECTOR = "cssSelector";
+  private static final String APP_NAME = "appName";
+  private static final String SCHEMA_NAME = "schemaName";
+
+  private final Database database;
+
+  public TriggerRepositoryImpl(Database database) {
+    this.database = database;
+    String currentUser = database.getActiveUser();
+    this.database.tx(
+        db -> {
+          try {
+            db.becomeAdmin();
+
+            Schema sysSchema = db.getSchema(SYSTEM_SCHEMA);
+            if (sysSchema == null) {
+              throw new MolgenisException("failed to setup Analytics TriggerRepository");
+            }
+
+            if (!sysSchema.getTableNames().contains(TRIGGER_TABLE_NAME)) {
+              sysSchema.create(
+                  table(
+                      TRIGGER_TABLE_NAME,
+                      column(NAME).setPkey(),
+                      column(CSS_SELECTOR).setRequired(true),
+                      column(APP_NAME).setRequired(false),
+                      column(SCHEMA_NAME).setRequired(false)));
+            }
+          } finally {
+            db.setActiveUser(currentUser);
+          }
+        });
+  }
+
+  @Override
+  public Trigger addTrigger(Trigger trigger) {
+    AtomicInteger inserted = new AtomicInteger();
+
+    this.database.tx(
+        db -> {
+          Schema sysSchema = db.getSchema(SYSTEM_SCHEMA);
+          if (sysSchema == null) {
+            throw new MolgenisException("failed to add analytics Trigger");
+          }
+          Table triggerTable = sysSchema.getTable(TRIGGER_TABLE_NAME);
+          Row triggerRow =
```

Comment:
interestingly I have been playing with idea to extend emx2 Table class to also accept `add(Object)` where the backend would automatically map records to rows.